File: Day-177(DSA-practice)/LinkedList.py
class LinkedList:
    def __init__(self):
        self.head = None  # initially empty list

    # Insert new node at end
    def insert(self, data):
        new_node = Node(data)

        if self.head is None:
            self.head = new_node
            return
        
        temp = self.head
        while temp.next:      # move till last node
            temp = temp.next
        temp.next = new_node  # attach new node

    # Print linked list
    def printList(self):
        temp = self.head
        while temp:
            print(temp.data, end=" -> ")
            temp = temp.next
        print("None")

File: Day-177(DSA-practice)/LinkedList.cpp
#include <bits/stdc++.h>
using namespace std;

struct Node {
    int data;
    Node* next;
};

int main() {
    Node* head = new Node(); // first node
    Node* second = new Node();
    Node* third = new Node();

    head->data = 10;
    head->next = second;

    second->data = 20;
    second->next = third;

    third->data = 30;
    third->next = NULL;

    // Traversal
    Node* temp = head;
    while (temp != NULL) {
        cout << temp->data << " ";
        temp = temp->next;
    }
}
